File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/interfaces/attribute/view/AttributeMainView.java
package com.nerdscorner.mvplib.testapp.interfaces.attribute.view;

import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.nerdscorner.mvplib.interfaces.view.BaseActivityView;
import com.nerdscorner.mvplib.testapp.R;
import com.nerdscorner.mvplib.testapp.interfaces.attribute.presenter.AttributeMainPresenterInterface;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class AttributeMainView extends BaseActivityView<AttributeMainPresenterInterface> implements AttributeMainViewInterface {
    @BindView(R.id.text) TextView textView;

    public AttributeMainView(AppCompatActivity activity) {
        super(activity);
        ButterKnife.bind(this, activity);
    }

    @OnClick(R.id.some_button)
    public void onActionClicked() {
        getPresenter().onActionClicked();
    }

    @Override
    public void setTextValue(CharSequence value) {
        textView.setText(value);
    }

}

File: testapp/src/main/java/com/github/marcherdiego/mvp/testapp/ui/mvp/view/AttributeMainView.kt
package com.github.marcherdiego.mvp.testapp.ui.mvp.view

import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.github.marcherdiego.mvp.events.bus.Bus
import com.github.marcherdiego.mvp.events.view.BaseActivityView
import com.github.marcherdiego.mvp.testapp.R

class AttributeMainView(activity: AppCompatActivity, bus: Bus) : BaseActivityView(activity, bus) {
    private var textView: TextView = activity.findViewById(R.id.text)

    init {
        onClick(R.id.some_button, ActionClickedEvent())
    }

    fun setTextValue(value: CharSequence) {
        textView.text = value
    }

    class ActionClickedEvent
}
